exchanges/stream/stream_match.go

```diff
@@ -5,76 +5,60 @@ import (
 	"sync"
 )
 
+var errSignatureCollision = errors.New("signature collision")
+
 // NewMatch returns a new Match
 func NewMatch() *Match {
-	return &Match{
-		m: make(map[interface{}]chan []byte),
-	}
+	return &Match{m: make(map[any]chan<- []byte)}
 }
 
 // Match is a distributed subtype that handles the matching of requests and
 // responses in a timely manner, reducing the need to differentiate between
 // connections. Stream systems fan in all incoming payloads to one routine for
 // processing.
 type Match struct {
-	m  map[interface{}]chan []byte
+	m  map[any]chan<- []byte
 	mu sync.Mutex
 }
 
-// Matcher defines a payload matching return mechanism
-type Matcher struct {
-	C   chan []byte
-	sig interface{}
-	m   *Match
-}
-
 // Incoming matches with request, disregarding the returned payload
-func (m *Match) Incoming(signature interface{}) bool {
+func (m *Match) Incoming(signature any) bool {
 	return m.IncomingWithData(signature, nil)
 }
 
 // IncomingWithData matches with requests and takes in the returned payload, to
 // be processed outside of a stream processing routine and returns true if a handler was found
-func (m *Match) IncomingWithData(signature interface{}, data []byte) bool {
+func (m *Match) IncomingWithData(signature any, data []byte) bool {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	ch, ok := m.m[signature]
-	if ok {
-		select {
-		case ch <- data:
-		default:
-			// this shouldn't occur but if it does continue to process as normal
-			return false
-		}
-		return true
+	if !ok {
+		return false
 	}
-	return false
+	ch <- data
+	close(ch)
+	delete(m.m, signature)
```

Comment:
Related to the comment below, we can't close it and delete it if we're allowing N+1